### ISSUE TITLE
constant ::Fixnum is deprecated warning with ruby 2.4.x

### ISSUE THREAD

**Issue by clairity** [hult/acts_as_having_string_id#15]
Ruby 2.4.x causes acts_as_having_string_id to display this warning on the console:
```
~/.rvm/gems/ruby-2.4.1@app/gems/acts_as_having_string_id-0.2.4/lib/acts_as_having_string_id/string_id.rb:60: warning: constant ::Fixnum is deprecated
```
due to unification of Fixnum and Bignum into Integer in Ruby 2.4.0: https://bugs.ruby-lang.org/issues/12005

From some quick testing based on http://blog.bigbinary.com/2016/11/18/ruby-2-4-unifies-fixnum-and-bignum-into-integer.html, I found that changing line 60 in `string_id.rb` to this:

```
if value.is_a?(String) || value.is_a?(1.class)
```
in place of 
```
if value.is_a?(String) || value.is_a?(Fixnum)
```
makes the warning go away.

**Comment by hult** [comment]
As I understand it, `Fixnum` inherits from `Integer` and 1 was previously a `Fixnum` but is, in 2.4, an `Integer`. The condition should probably be `value <= Integer` (to catch `Bignum` in rubies before 2.4). Will include in next release!

**Comment by hult** [comment]
I released 1.2.5 which works fine for me. Thanks for reporting!
